### PR TITLE
prepare IW code to be enabled in web without actually enabling anything yet

### PR DESCRIPTION
**Pull request by amunger**
removed direct fspath usage and split files for some of the harder features to translate to web.
This is just a pausing point as I shift to working on enabling IW in web in the core.

### CODE REVIEW

File: src/interactive-window/interactiveWindowCommandListener.ts [hunk 1]
```diff
@@ -216,21 +218,21 @@ export class InteractiveWindowCommandListener implements IDataScienceCommandList
                                     directoryChange = uri;
                                 }
 
-                                const notebook = await this.jupyterExporter.translateToNotebook(
+                                const notebook = await this.jupyterExporter?.translateToNotebook(
```
Comment (IanMatthewHuff) [inline src/interactive-window/interactiveWindowCommandListener.ts:221]: Could we check the existance of this earlier? Seems like we're doing work (showing the dialog and stuff) when we might not actually have the ability to export.

File: src/interactive-window/interactiveWindowCommandListener.ts [hunk 2]
```diff
@@ -348,7 +351,12 @@ export class InteractiveWindowCommandListener implements IDataScienceCommandList
 
     @captureTelemetry(Telemetry.CreateNewInteractive, undefined, false)
     private async createNewInteractiveWindow(connection?: KernelConnectionMetadata): Promise<void> {
-        await this.interactiveWindowProvider.getOrCreate(undefined, connection);
+        try {
+            await this.interactiveWindowProvider.getOrCreate(undefined, connection);
+        } catch (error) {
```
Comment (IanMatthewHuff) [inline src/interactive-window/interactiveWindowCommandListener.ts:356]: Not that this hurts anything, but just interested. Don't errors get logged anyways? Was this fixing some specific case where an error was getting swallowed?

Comment (amunger) [reply src/interactive-window/interactiveWindowCommandListener.ts:356]: whoops, this was just for debugging, didn't mean to leave this in

File: src/interactive-window/interactiveWindowProvider.ts [hunk 3]
```diff
@@ -224,7 +224,7 @@ export class InteractiveWindowProvider implements IInteractiveWindowProvider, IA
             !this.globalMemento.get(AskedForPerFileSettingKey) &&
             this._windows.length === 1 &&
             // Only prompt if the submitting file is different
-            this._windows[0].owner?.fsPath !== resource.fsPath
+            this._windows[0].owner?.path !== resource.path
```
Comment (IanMatthewHuff) [inline src/interactive-window/interactiveWindowProvider.ts:227]: Does this line need to use !arePathsSame? Or intended as a bare comparison.

**Review by rchiodo — APPROVED** [review]

File: src/platform/terminals/codeExecution/codeExecutionHelper.ts [hunk 4]
```diff
@@ -7,74 +7,21 @@ import { Position, Range, TextEditor, Uri } from 'vscode';
...
-import { IProcessServiceFactory } from '../../common/process/types.node';
-import { createDeferred } from '../../common/utils/async';
-import { IInterpreterService } from '../../interpreter/contracts';
 import { IServiceContainer } from '../../ioc/types';
 import { ICodeExecutionHelper } from '../types';
 
 @injectable()
 export class CodeExecutionHelper implements ICodeExecutionHelper {
-    private readonly documentManager: IDocumentManager;
+    protected readonly documentManager: IDocumentManager;
```
[the remark this answers is not in the document]
Comment (amunger) [reply src/platform/terminals/codeExecution/codeExecutionHelper.ts:15]: this was the shared code, and the .node version overrides with the additional features. Unfortunately our DI test doesn't allow for injectables to be used like this, so I'll just make a base class and have .web and .node extensions.

File: src/interactive-window/interactiveWindow.ts [hunk 5]
```diff
@@ -819,7 +819,7 @@ export class InteractiveWindow implements IInteractiveWindowLoadable {
         let defaultFileName;
         if (this.submitters && this.submitters.length) {
             const lastSubmitter = this.submitters[this.submitters.length - 1];
-            defaultFileName = path.basename(lastSubmitter.fsPath, path.extname(lastSubmitter.fsPath));
+            defaultFileName = path.basename(lastSubmitter.path, path.extname(lastSubmitter.path));
```
[the remark this answers is not in the document]
Comment (amunger) [reply src/interactive-window/interactiveWindow.ts:822]: The only difference that I can see is how the drive is specified (prefixed with a '/'), pulling out filenames seems to work fine, at least with my manual testing on my Windows machine.

File: src/interactive-window/interactiveWindowCommandListener.ts (hunk 1 above)
Comment (amunger) [reply src/interactive-window/interactiveWindowCommandListener.ts:221]: sure, though I still need this check for typescript to be happy

**Review by IanMatthewHuff — APPROVED** [review]

**Review by DonJayamanne — APPROVED** [review]

**Pull request merged**